enterprise/core-edge/src/main/java/org/neo4j/coreedge/catchup/storecopy/CoreClient.java

```diff
@@ -164,6 +181,12 @@ public void onTxReceived( final TxPullResponse tx )
     }
 
     @Override
+    public void onStoreIdReceived( final StoreId storeId )
```

Comment:
This API looks strange and fragile. If `setStoreIdConsumer` was not called than this method will throw NPE. I think `requestStoreId` should just return a `Future`.